spec/api_spec.rb

```diff
@@ -21,10 +21,10 @@ class Bar; end
       end
 
       context "when setting custom options" do
-        before { subject.setup foo: { bar: "baz" }, url: "https://api.example.com" }
+        before { subject.setup foo: { bar: "baz" }, url: "https://api.example.com", timeout: 10, open_timeout: 2 }
```

Comment:
Sorry, still haven't tested it. But found another connected bug: [Net::HTTP retries idempotent requests once after a timeout, but its not configurable](https://bugs.ruby-lang.org/issues/10674). So if we will fix this bug, and I will pass timeout, Net::HTTP will do two requests in case of Timeout. Nice.